src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +591,104 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            ' only exists for the convenience of SemanticModel, so we want to classify it as a Delegate Creation
+            Dim validConversionKind = conversionKind = ConversionKind.Identity OrElse
+                                      (conversionKind And (Not ConversionKind.DelegateRelaxationLevelMask)) = 0
+
+            ' Any of the explicit cast types, as well as New DelegateType(AddressOf Method)
+            Dim validConversionSyntax = conversionSyntax.Kind() = SyntaxKind.CTypeExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.DirectCastExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.TryCastExpression OrElse
+                                        conversionSyntax.Kind() = SyntaxKind.ObjectCreationExpression
+
+            Dim validOperandSyntax = operandSyntax.Kind() = SyntaxKind.AddressOfExpression
+
+            Return validConversionKind AndAlso validConversionSyntax AndAlso validOperandSyntax
+        End Function
+
+        ''' <summary>
+        ''' Creates the Lazy IOperation from a delegate creation operand or a bound conversion operand, handling when the conversion
+        ''' is actually a delegate creation.
+        ''' </summary>
+        Private Function CreateConversionOperation(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode
+                                                   ) As (methodSymbol As MethodSymbol, operation As Lazy(Of IOperation), isAddressOfDelegateCreation As Boolean)
+            If (conversionKind And VisualBasic.ConversionKind.UserDefined) = VisualBasic.ConversionKind.UserDefined Then
+                Dim userDefinedConversion As BoundUserDefinedConversion = DirectCast(operand, BoundUserDefinedConversion)
+                Return (userDefinedConversion.Call.Method, New Lazy(Of IOperation)(Function() Create(userDefinedConversion.Operand)), isAddressOfDelegateCreation:=False)
+            ElseIf IsDelegateCreationSyntax(conversionKind, conversionSyntax, operand.Syntax) Then
+                ' In this scenario, we're a delegate creation expression involving an AddressOf.
+                Dim methodSymbol As MethodSymbol = Nothing
+                Dim operandLazy As Lazy(Of IOperation)
+                If operand.Kind = BoundKind.DelegateCreationExpression Then
+                    operandLazy = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression)))
+                Else
+                    ' This is an error scenario in which we have a delegate conversion, but it failed to bind correctly, so the child here
+                    ' is either a bad expression or a method group. Delegate to standard operation handling for this.
+                    Debug.Assert(operand.Kind = BoundKind.MethodGroup OrElse
```

Comment:
>operand.Kind = BoundKind.MethodGroup [](start = 33, length = 36)

Could you point me a to a test case, for which this condition is true?

Reply:
I believe this was true earlier, but in later revisions this part of the condition is never hit.

As part of the CreateConversionOperand refactor I just did, this branch is no longer just for exceptional code, and can be hit by lambdas as well. Because of this, I've removed the assert altogether.

---
In reply to: [142995679](https://github.com/dotnet/roslyn/pull/22056#discussion_r142995679) [](ancestors = 142995679)